smarty/environment.py

```diff
@@ -128,27 +162,29 @@ def addANDtype(self, ANDtype):
             ANDtype: string
                 added to the list of ANDtypes for this atom
             """
-            self.ANDtypes.append(ANDtype)
+            if ANDtype != "":
```

Comment:
yes, I updated this and updated a few other places to guarantee no repeated decorators